internal/fs/async_fetch_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/fs"
+	"github.com/googlecloudplatform/gcsfuse/internal/fs/inode"
+	"github.com/googlecloudplatform/gcsfuse/internal/gcsx"
+	"github.com/jacobsa/fuse/fuseops"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsfake"
+	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
+	"golang.org/x/net/context"
+)
+
+const dirInodeID = 17
+const dirInodeName = "foo/"
+const dirMode os.FileMode = 0712 | os.ModeDir
+const typeCacheTTL = time.Second
+const uid = 123
+const gid = 456
+
+type AsyncFetchTest struct {
+	ctx    context.Context
+	bucket gcsx.SyncerBucket
+	clock  timeutil.SimulatedClock
+	dh     *fs.DirHandle
+}
+
+var _ SetUpInterface = &AsyncFetchTest{}
+var _ TearDownInterface = &AsyncFetchTest{}
+
+func init() {
+	RegisterTestSuite(&AsyncFetchTest{})
+}
+
+func (t *AsyncFetchTest) SetUp(ti *TestInfo) {
+	t.ctx = ti.Ctx
+	t.clock.SetTime(time.Date(2015, 4, 5, 2, 15, 0, 0, time.Local))
+	bucket := gcsfake.NewFakeBucket(&t.clock, "some_bucket")
+	t.bucket = gcsx.NewSyncerBucket(
+		1, // Append threshold
+		".gcsfuse_tmp/",
+		bucket)
+	// Create the inode and directory handle. No implicit dirs by default.
```

Comment:
I think this comment does not belongs to this place. Please put it in right place. e.g. above appropriate function